unliftio/test/UnliftIO/PooledAsyncSpec.hs

```diff
@@ -0,0 +1,174 @@
+module UnliftIO.PooledAsyncSpec (spec) where
+
+import Test.Hspec
+import Control.Concurrent
+import Data.List (nub, sort)
+import Test.QuickCheck
+import Data.Functor ((<$>))
+import UnliftIO
+
+spec :: Spec
+spec = do
+  let exAction :: Int -> IO Int
+      exAction x = do
+        if (x == 2) then error "hell" else return ()
```

Comment:
I usually try to create custom exceptions for test cases, this way I am sure I am getting the exact error I am expecting. You never know for sure if you are getting your `error "hell"` or some totally different `error` from somewhere else down the call tree. It's not terribly important here, so you can leave it as is if you like.

Reply:
Done.